### PR TITLE
feat/pitch interp

### PR DESCRIPTION
**Pull request by mensshi**
adds linear interpolation for NaNs
+ functions for visualization

### CODE REVIEW

File: tonami/pitch_process.py [hunk 1]
```diff
@@ -96,10 +96,52 @@ def get_voice_activity(pitch_contour):
...
+        index: a function that takes an array with 1s representing NaNs, 
+            and returns the indexes of nans
+    Example:
+        >>> # linear interpolation of NaNs
+        >>> arr = np.array([3,np.nan,4,np.nan,5)
+
+        >>> nans, index= nan_helper(arr)
+        >>> # nans would be [0, 1, 0, 1, 0]
+        >>> # index(nans) would return [1, 3]
+        >>> arr[nans]= np.interp(index(nans), index(~nans), arr[~nans])
```
Comment (saltyseadawg) [inline tonami/pitch_process.py:119]: might as well convert this fully to a doctest format (it's basically 90% of the way there)

https://docs.python.org/3/library/doctest.html

File: data_viz/viz.py [hunk 2]
```diff
@@ -1,20 +1,46 @@
...
 import pandas as pd
+from tonami import pitch_process as pp
 
 # helper/utility functions specifically for plotting
 
+PITCH_FILEPATH = 'data/parsed/toneperfect_pitch_librosa_50-500-fminmax.json'
 
-def plot_tones(tones):
+# currently, filename specifications only work for the above PITCH_FILEPATH
+def plot_tones(filename='wo3_MV2_MP3.mp3'):
```
Comment (saltyseadawg) [inline data_viz/viz.py]: typehint pls

File: data_viz/viz.py [hunk 3]
```diff
@@ -1,20 +1,46 @@
...
     """
-    Plots the list of tones.
+    Plots the specified tones.
 
     Args:
-        tones: a list/array of tones to be plotted.
+        filename: filename of tone to be plotted
```
Comment (saltyseadawg) [inline data_viz/viz.py]: typehint pls

File: data_viz/viz.py [hunk 4]
```diff
@@ -1,20 +1,46 @@
...
+    pitch_data = pitch_data.loc[pitch_data['filename'].isin([filename])]
 
-    pass
+    #plot before and after interp
+    pitch_contour = pitch_data.loc[:, 'pitch_contour'].to_numpy()
+    plt.plot(pitch_contour[0])
+    plt.savefig('tone_' + filename + ".jpg")
 
+# could probably be combined with above at some point
+def visualize_interp(filename='wo3_MV2_MP3.mp3'):
```
Comment (saltyseadawg) [inline data_viz/viz.py]: typehint pls

File: data_viz/viz.py [hunk 5]
```diff
@@ -1,20 +1,46 @@
...
+    """
+    Plots the specified tones, and a version that has been interpolated
+
+    Args:
+        filename: filename of tone to be plotted
```
Comment (saltyseadawg) [inline data_viz/viz.py]: typehint pls 
i.e. filename (str): xyz

File: tonami/pitch_process.py [hunk 6]
```diff
@@ -96,10 +96,52 @@ def get_voice_activity(pitch_contour):
 
     return voiced
 
+def get_nan_idx(arr):
```
Comment (saltyseadawg) [inline tonami/pitch_process.py]: typehints for params and returns

File: tonami/pitch_process.py [hunk 7]
```diff
@@ -96,10 +96,52 @@ def get_voice_activity(pitch_contour):
...
+    """
+    Helper to handle indices and logical indices of NaNs.
+
+    Args:
+        arr: 1d numpy array with possible NaNs
+    Returns:
+        nans: logical indices of NaNs (an array of the same size as arr, 
+            which marks the location of NaNs with 1, and all others with 0)
+        index: a function that takes an array with 1s representing NaNs, 
+            and returns the indexes of nans
```
Comment (saltyseadawg) [inline tonami/pitch_process.py]: typehints

File: data_viz/viz.py (hunk 3 above)
Comment (saltyseadawg) [inline data_viz/viz.py]: - plot_pitch_contour() might be a better function name -> more explicit, plot_tone is kind of ambiguous about what tone features we might be displaying
- by the same token, would be helpful to describe what kind of plot is being generated in the description

File: tonami/pitch_process.py (hunk 1 above)
Comment (mensshi) [reply tonami/pitch_process.py:119]: lol

**Pull request merged**